solidity/contracts/vault/DonationVault.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.9;
+
+import "./IVault.sol";
+import "../bank/Bank.sol";
+
+/// @title BTC donation vault
+/// @notice Vault that allows making BTC donations to the system. Upon deposit,
+///         this vault does not increase depositors' tBTC balances and always
+///         decreases its own tBTC balance in the same transaction.
+///
+///         BEWARE: ALL BTC DEPOSITS TARGETING THIS VAULT ARE NOT REDEEMABLE AND
+///         THERE IS NO WAY TO RESTORE THE TBTC BALANCE! DO NOT DEPOSIT AGAINST
+///         THIS VAULT UNLESS YOU REALLY KNOW WHAT YOU ARE DOING!
+contract DonationVault is IVault {
+    Bank public bank;
+
+    event DonationReceived(address[] depositors, uint256[] depositedAmounts);
+
+    modifier onlyBank() {
+        require(msg.sender == address(bank), "Caller is not the Bank");
+        _;
+    }
+
+    constructor(Bank _bank) {
+        require(
+            address(_bank) != address(0),
+            "Bank can not be the zero address"
+        );
+
+        bank = _bank;
+    }
+
+    /// @notice Function not supported by the `DonationVault`. Always reverts.
+    function receiveBalanceApproval(address, uint256)
+        external
+        override
+        onlyBank
+    {
+        revert("Donation vault cannot receive balance approval");
```

Comment:
Why not to allow someone make a donation from their Bank balance? 

I think we should allow it by `Bank.approveBalanceAndCall` as well as implement `DonationVault.donate()` for already approved balances.

Reply:
Yep, good idea! Here it is: https://github.com/keep-network/tbtc-v2/pull/269/commits/4f2b9d0bcc28ff55a08e12881d2c28331b3000e0